plugins/inputs/statsd/statsd.go

```diff
@@ -636,20 +742,115 @@ func (s *Statsd) aggregate(m metric) {
 	}
 }
 
+// handler handles a single TCP Connection
+func (s *Statsd) handler(conn *net.TCPConn, id string) {
+	s.CurrentConnections.Incr(1)
+	s.TotalConnections.Incr(1)
+	// connection cleanup function
+	defer func() {
+		s.wg.Done()
+		conn.Close()
+		// Add one connection potential back to channel when this one closes
+		s.accept <- true
+		s.forget(id)
+		s.CurrentConnections.Incr(-1)
+	}()
+
+	var n int
+	scanner := bufio.NewScanner(conn)
+	for {
+		select {
+		case <-s.done:
+			return
+		default:
+			if !scanner.Scan() {
+				return
+			}
+			n = len(scanner.Bytes())
+			if n == 0 {
+				continue
+			}
+			s.BytesRecv.Incr(int64(n))
+			s.PacketsRecv.Incr(1)
+			bufCopy := make([]byte, n+1)
+			copy(bufCopy, scanner.Bytes())
+			bufCopy[n] = '\n'
+
+			select {
+			case s.in <- bufCopy:
+			default:
+				s.drops++
+				if s.drops == 1 || s.drops%s.AllowedPendingMessages == 0 {
+					log.Printf(dropwarn, s.drops)
+				}
+			}
+		}
+	}
+}
+
+// refuser refuses a TCP connection
+func (s *Statsd) refuser(conn *net.TCPConn) {
+	// Tell the connection why we are closing.
+	fmt.Fprintf(conn, "Telegraf maximum concurrent TCP connections (%d)"+
+		" reached, closing.\nYou may want to increase max_tcp_connections in"+
+		" the Telegraf tcp listener configuration.\n", s.MaxTCPConnections)
```

Comment:
I am assuming this is not part of the statsd protocol, any chance this will confuse a tcp statsd implementation?  Maybe we should just close the connection?